api-channel-jira/src/main/java/com/synopsys/integration/alert/api/channel/jira/distribution/custom/MessageReplacementValues.java

```diff
@@ -82,4 +70,73 @@ public String getProjectName() {
         return Optional.ofNullable(policyCategory);
     }
 
+    public static class Builder {
+        private String providerName;
+        private String projectName;
+        private String projectVersionName;
+        private String componentName;
+        private String componentVersionName;
+        private String severity;
+        private String policyCategory;
+
+        public Builder(String providerName, String projectName) {
+            this.providerName = providerName;
+            this.projectName = projectName;
+        }
+
+        public MessageReplacementValues build() {
+            return new MessageReplacementValues(
+                providerName,
+                projectName,
+                defaultIfBlank(projectVersionName),
+                defaultIfBlank(componentName),
+                defaultIfBlank(componentVersionName),
+                defaultIfBlank(severity),
+                defaultIfBlank(policyCategory)
+            );
+        }
+
+        public Builder providerName(String providerName) {
+            this.providerName = providerName;
+            return this;
+        }
+
+        public Builder projectName(String projectName) {
+            this.projectName = projectName;
+            return this;
+        }
+
+        public Builder projectVersionName(String projectVersionName) {
+            this.projectVersionName = projectVersionName;
+            return this;
+        }
+
+        public Builder componentName(String componentName) {
+            this.componentName = componentName;
+            return this;
+        }
+
+        public Builder componentVersionName(String componentVersionName) {
+            this.componentVersionName = componentVersionName;
+            return this;
+        }
+
+        public Builder severity(String severity) {
+            this.severity = severity;
+            return this;
+        }
+
+        public Builder policyCategory(String policyCategory) {
+            this.policyCategory = policyCategory;
+            return this;
+        }
+
+        private String defaultIfBlank(String replacementField) {
+            if (!StringUtils.isBlank(replacementField)) {
```

Comment:
`StringUtils` has an `isNotBlank(...)` method as well. I think that would be slightly more readable here. Or we could remove the negation and swap the returns. Any time we can avoid negation is a win for readability.